packages/render/src/operations/transform.js

```diff
@@ -2,113 +2,133 @@ import * as R from 'ramda';
-  const matchX = /translateX\((-?\d+\.?d*)\)/g.exec(transform);
-  const matchGeneric = /translate\((-?\d+\.?d*).*(,|\s)\s*(-?\d+\.?d*).*\)/g.exec(
-    transform,
-  );
-
-  if (matchX && matchX[1]) return matchX[1];
-  if (matchGeneric && matchGeneric[1]) return matchGeneric[1];
+  const parsed = [];
 
-  return 0;
-};
-
-const getTranslateY = transform => {
-  const matchY = /translateY\((-?\d+\.?\d*)\)/g.exec(transform);
-  const matchGeneric = /translate\((-?\d+\.?\d*).*(,|\s)\s*(-?\d+\.?\d*).*\)/g.exec(
-    transform,
-  );
+  for (let i = 0; i < transforms.length; i += 1) {
+    const transform = transforms[i];
 
-  if (matchY && matchY[1]) return matchY[1];
-  if (matchGeneric && matchGeneric[3]) return matchGeneric[3];
+    if (transform) {
+      const [operationName, operationValueRaw] = transform.split('(');
+      const values = operationValueRaw.split(',').map(val => val.trim());
+      const value = values.length === 1 ? values[0] : values;
+      parsed.push([operationName, value]);
+    }
+  }
 
-  return 0;
+  return parsed;
 };
 
-const getScaleX = transform => {
-  const matchX = /scaleX\((-?\d+\.?\d*)\)/g.exec(transform);
-  const matchGeneric = /scale\((-?\d+\.?\d*).*,?\s*(-?\d+\.?\d*).*\)/g.exec(
-    transform,
-  );
-
-  if (matchX && matchX[1]) return matchX[1];
-  if (matchGeneric && matchGeneric[1]) return matchGeneric[1];
-
-  return 1;
+const applySingleTransformation = (ctx, operation, origin) => {
+  const [name, args] = operation;
+
+  switch (name) {
+    case 'scale': {
+      const [scaleX, scaleY] = args;
+      ctx.scale(scaleX, scaleY, { origin });
+      break;
+    }
+
+    case 'rotate': {
+      ctx.rotate(args, { origin });
+      break;
+    }
+
+    case 'translate': {
+      const [x, y] = args;
+      ctx.translate(x, y, { origin });
+      break;
+    }
+
+    case 'matrix': {
+      ctx.transform(...args);
+      break;
+    }
+
+    default: {
+      console.error(`Transform operation ${name} doesn't supported`);
+    }
+  }
 };
 
-const getScaleY = transform => {
-  const matchY = /scaleY\((-?\d+\.?\d*)\)/g.exec(transform);
-  const matchGeneric = /scale\((-?\d+\.?\d*).*,?\s*(-?\d+\.?\d*).*\)/g.exec(
-    transform,
-  );
-
-  if (matchY && matchY[1]) return matchY[1];
-  if (matchGeneric && matchGeneric[2]) return matchGeneric[2];
-
-  return 1;
+const normalizeOperation = (operation, args) => {
+  switch (operation) {
+    case 'scale': {
+      const isArray = Array.isArray(args);
+      const parsed = isArray
+        ? [Number.parseFloat(args[0]), Number.parseFloat(args[1])]
+        : Number.parseFloat(args);
+      const [scaleX, scaleY] = isArray ? parsed : [parsed, parsed];
+      return ['scale', [scaleX, scaleY]];
+    }
+
+    case 'scaleX': {
+      return ['scale', [Number.parseFloat(args), 1]];
+    }
+    case 'scaleY': {
+      return ['scale', [1, Number.parseFloat(args)]];
+    }
+
+    case 'rotate': {
+      const unitsRegexp = /(-?\d*\.?\d*)(\w*)?/i;
```

Comment:
what about parsing and converting units? this should work as before, but it can be also `turn` and `grad`

does `@react-pdf` have util for parsing unit that i can use here?

Reply:
No, since all unit transformations happen in the `@react/stylesheet` package. That makes me think if it shouldn't be better to move that logic to there. See comment below